src/algebra/module/localized_module.lean

```diff
@@ -387,4 +387,280 @@ instance localized_module_is_localized_module :
+  (h s).unit⁻¹.1 (g m) := rfl
+
+lemma _root_.localized_module.lift'_add (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) (x y) :
+  localized_module.lift' S g h (x + y) =
+  localized_module.lift' S g h x + localized_module.lift' S g h y :=
+localized_module.induction_on₂ begin
+  intros a a' b b',
+  erw [localized_module.lift'_mk, localized_module.lift'_mk, localized_module.lift'_mk],
+  dsimp, generalize_proofs h1 h2 h3,
+  erw [map_add, module.End_algebra_map_is_unit_inv_apply_eq_iff,
+    smul_add, ←h2.unit⁻¹.1.map_smul, ←h3.unit⁻¹.1.map_smul],
+  congr' 1; symmetry,
+  erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, mul_smul, ←map_smul], refl,
+  erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, mul_comm, mul_smul, ←map_smul], refl,
+end x y
+
+lemma _root_.localized_module.lift'_smul (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x))
+  (r : R) (m) :
+  r • localized_module.lift' S g h m = localized_module.lift' S g h (r • m) :=
+m.induction_on begin
+  intros a b,
+  rw [localized_module.lift'_mk, localized_module.smul'_mk, localized_module.lift'_mk],
+    generalize_proofs h1 h2,
+    erw [←h1.unit⁻¹.1.map_smul, ←g.map_smul],
+end
+
+/--
+If `g` is a linear map `M → M''` such that all scalar multiplication by `s : S` is invertible, then
+there is a linear map `localized_module S M → M''`.
+-/
+noncomputable def _root_.localized_module.lift (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) :
+  (localized_module S M) →ₗ[R] M'' :=
+{ to_fun := localized_module.lift' S g h,
+  map_add' := localized_module.lift'_add S g h,
+  map_smul' := λ r x, by rw [localized_module.lift'_smul, ring_hom.id_apply] }
+
+lemma _root_.localized_module.lift_mk (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x))
+  (m : M) (s : S) :
+  localized_module.lift S g h (localized_module.mk m s) = (h s).unit⁻¹.1 (g m) := rfl
+
+lemma _root_.localized_module.lift_comp (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) :
+  (localized_module.lift S g h).comp (localized_module.mk_linear_map S M) = g :=
+begin
+  ext x, dsimp, rw localized_module.lift_mk,
+  erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, one_smul],
+end
+
+lemma _root_.localized_module.lift_unique (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x))
+  (l : localized_module S M →ₗ[R] M'')
+  (hl : l.comp (localized_module.mk_linear_map S M) = g) :
+  localized_module.lift S g h = l :=
+begin
+  ext x, induction x using localized_module.induction_on with m s,
+  rw [localized_module.lift_mk],
+  erw [module.End_algebra_map_is_unit_inv_apply_eq_iff, ←hl, linear_map.coe_comp, function.comp_app,
+    localized_module.mk_linear_map_apply, ←l.map_smul, localized_module.smul'_mk],
+  congr' 1, rw localized_module.mk_eq,
+  refine ⟨1, _⟩, simp only [one_smul], refl,
+end
```

Comment:
Just a minor comment, but I would be inclined to move this to line 351 into the `localized_module` so that `_root_` could be avoided.

Reply:
done in [9d95a73](https://github.com/leanprover-community/mathlib/pull/15559/commits/9d95a73c230ebfc3c12e4b8e583698f1b4960505)